samples/bluetooth/iso_broadcast_benchmark/src/receiver.c

```diff
@@ -0,0 +1,510 @@
+		BT_INTERVAL_TO_MS(biginfo->iso_interval),
+		biginfo->burst_number, biginfo->offset, biginfo->rep_count,
+		biginfo->max_pdu, biginfo->sdu_interval, biginfo->max_sdu,
+		phy2str(biginfo->phy), biginfo->framing ? "with" : "without",
+		biginfo->encryption ? "" : "not ");
+
+	iso_interval_ms = BT_INTERVAL_TO_MS(biginfo->iso_interval);
+	bis_count = MIN(biginfo->num_bis, CONFIG_BT_ISO_MAX_CHAN);
+	biginfo_received = true;
+	k_sem_give(&sem_per_big_info);
+}
+
+static struct bt_le_per_adv_sync_cb sync_callbacks = {
+	.synced = sync_cb,
+	.term = term_cb,
+	.biginfo = biginfo_cb,
+};
+
+static void print_stats(char *name, struct iso_recv_stats *stats)
+{
+	/* Convert to a two decimal value without using floats, by
+	 * multiplying with 10000 before division, and then seperating
+	 * the lowest 2 digits
+	 */
+	uint8_t integer;
+	uint8_t fraction;
+	uint32_t total_packets;
+	uint64_t tmp;
+
+	total_packets = stats->iso_recv_count + stats->iso_lost_count;
+	tmp = ((uint64_t)stats->iso_recv_count * 10000) / total_packets;
+
+	integer = tmp / 100;
+	fraction = tmp - integer * 100;
+
+	LOG_INF("%s: Received %u/%u (%u.%02u%%) - Total packets lost %u",
+		name, stats->iso_recv_count, total_packets, integer, fraction,
+		stats->iso_lost_count);
+
+}
+
+static void iso_recv(struct bt_iso_chan *chan, struct net_buf *buf)
+{
+	uint32_t counter = sys_get_le32(buf->data);
+	uint32_t total_packets;
+	static bool latest_stats_arr[1000];
+	static size_t latest_stats_arr_pos;
+	uint32_t packets_lost;
+
+	/* NOTE: The packets received may be on different BISes */
+
+	if (buf->len == 0) {
+		LOG_INF("Data length is zero - Bad packet");
+		return;
+	}
+
+	if (counter <= last_received_counter) {
+		LOG_INF("Counter (%u) was less or to last counter received (%u)",
+			counter, last_received_counter);
+		return;
+	}
+
+	packets_lost = counter - last_received_counter - 1;
+
+	if (packets_lost > 0 && last_received_counter > 0) {
+
+		current_sync_stats.iso_lost_count += packets_lost;
+		overall_stats.iso_lost_count += packets_lost;
+
+		for (int i = 0; i < packets_lost; i++) {
+			latest_stats_arr[latest_stats_arr_pos++] = false;
+			if (latest_stats_arr_pos == sizeof(latest_stats_arr)) {
+				latest_stats_arr_pos = 0;
+			}
+		}
+	}
+
+	current_sync_stats.iso_recv_count++;
+	overall_stats.iso_recv_count++;
+
+	latest_stats_arr[latest_stats_arr_pos++] = true;
+	if (latest_stats_arr_pos == sizeof(latest_stats_arr)) {
+		latest_stats_arr_pos = 0;
+	}
+
+	last_received_counter = counter;
+
+	total_packets = overall_stats.iso_recv_count + overall_stats.iso_lost_count;
+
+	if ((total_packets % 100) == 0) {
+		struct iso_recv_stats latest_stats = { 0 };
+
+		for (int i = 0; i < ARRAY_SIZE(latest_stats_arr); i++) {
+			if (i == total_packets) {
+				break;
+			}
+
+			if (latest_stats_arr[i]) {
+				latest_stats.iso_recv_count++;
+			} else {
+				latest_stats.iso_lost_count++;
+			}
+		}
+
+		print_stats("Overall     ", &overall_stats);
```

Comment:
Suggest renaming to:
`stats_overall`, `stats_latest` and `stats_current_sync`. Makes it more readable.

Reply:
The string values, or the variable names? :)